### PR TITLE
[#108441228] Add container for AWS cli tool

### PR DESCRIPTION
**Pull request by keymon**
[#108441228 Automatically shutdown dev environments at night / weekends](https://www.pivotaltracker.com/story/show/108441228)

What
----

We want to be able to stop the vagrant AWS vm at nights. We will use AWS
cli tools to send the command to terminate it.

Scope
-----

Add a container including the AWS client tools. We create it based on
the curl container, so that we can use curl for querying the metadata.

How to test this?
-----------------

Run the tests:

```
bundle install # I recommend rvm
rake build:awscli spec:awscli
```

Run the container:

```
docker run awscli aws --version
```

Who?
----
Anyone but @keymon or @actionjack

### CODE REVIEW

File: awscli/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+It also includes the ca-certificates.
+
+Based on [alpine](https://hub.docker.com/_/alpine/) image.
+
+## Build locally
+
+```
+$ cd bosh-init
+$ docker build -t curl-ssl .
```
Comment (alext) [inline awscli/README.md]: This refers to `bosh-init` and `curl-ssl` instead of `awscli`.  Also below...

**Comment by saliceti** [comment]
`aws help` fails. It's not critical but can be helpful.

The additional required packages are groff and less, it takes 8MB additional space.

File: awscli/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Also curl with SSL and openssl support is included, to allow scripting HTTP
+calls if necessary (e.g. query metadata)
+
+Based on [alpine](https://hub.docker.com/_/alpine/) image.
+
+## Build locally
+
+```
+$ cd bosh-init
```
Comment (combor) [inline awscli/README.md]: `cd awscli` ?

**Comment by saliceti** [comment]
Tested successfully:

```
$ docker run -ti -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID awscli aws ec2 describe-instances --region eu-west-1
{
    "Reservations": [
        {
            "OwnerId": "988997429095", 
            "ReservationId": "r-952e1f34", 
            "Groups": [], 
...
```
I will test if it works with instance profile in: https://github.com/alphagov/paas-cf/pull/59

**Comment by keymon** [comment]
Fixed the README and added support for the help.

**Pull request merged**